endpoints/api/tag.py

```
@@ -197,6 +206,40 @@ def put(self, namespace, repository, tag):
             else:
                 raise InvalidRequest("Could not update tag expiration; Tag has probably changed")
 
+        if "immutable" in request.get_json():
+            if not features.IMMUTABLE_TAGS:
+                raise InvalidRequest("Tag immutability is not enabled")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)